x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/schemas/query_signals_index_schema.ts

```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Joi from 'joi';
+
+export const querySignalsSchema = Joi.object({
+  search_query: Joi.object().required(),
+});
```

Comment:
If you dropped the `search_query`, you could change this to just be that `query` is required? Unless you were planning more than just `query` to be accepted.

Reply:
If you wanted aggregations too you could then do a "at least one of these is required" with the top level keys being:

```ts
query
aggregations
```

Just kind of asking since I have to do these type of endpoints myself for `rules`

Reply:
Ah yeah I forgot about aggregations. Good call.

Reply:
Yeah, ++ on aggregations -- we'll want to use this for the Signals histogram.